src/main/java/com/conveyal/datatools/common/utils/Scheduler.java

```diff
@@ -111,71 +129,93 @@ public static void removeFeedSourceJobsOfType(String id, Class<?> clazz, boolean
     }
 
     /**
-     * Cancels and removes all scheduled jobs for a given project id and job class.
+     * Cancels and removes all scheduled jobs for a given feed source id (of any job type).
      */
-    public static void removeProjectJobsOfType(String id, Class<?> clazz, boolean mayInterruptIfRunning) {
-        int cancelled = removeJobsOfType(scheduledJobsForProjects, id, clazz, mayInterruptIfRunning);
-        if (cancelled > 0) LOG.info("Cancelled/removed {} {} jobs for project {}", cancelled, clazz.getSimpleName(), id);
+    public static void removeAllFeedSourceJobs(String id, boolean mayInterruptIfRunning) {
+        int cancelled = removeAllJobs(scheduledJobsForFeedSources, id, mayInterruptIfRunning);
+        if (cancelled > 0) LOG.info("Cancelled/removed {} jobs for feed source {}", cancelled, id);
     }
 
     /**
-     * Schedule or cancel auto feed fetch for a project as needed.  This should be called whenever a
-     * project is created or updated.  If a project is deleted, the auto feed fetch jobs will
+     * Schedule or cancel auto feed fetch for a project's feeds as needed.  This should be called whenever a
+     * project is created or updated.  If a feed source is deleted, the auto feed fetch jobs will
      * automatically cancel itself.
      */
-    public static void scheduleAutoFeedFetch(Project project) {
-        // If auto fetch flag is turned on, schedule auto fetch.
-        if (project.autoFetchFeeds) Scheduler.scheduleAutoFeedFetch(project, 1);
-        // Otherwise, cancel any existing task for this id.
-        else Scheduler.removeProjectJobsOfType(project.id, FetchProjectFeedsJob.class, true);
+    public static void handleAutoFeedFetch(Project project) {
+        long defaultDelay = getDefaultDelayMinutes(project);
+        for (FeedSource feedSource : project.retrieveProjectFeedSources()) {
+            scheduleAutoFeedFetch(feedSource, defaultDelay);
+        }
+    }
+
+    private static long getDefaultDelayMinutes(Project project) {
```

Comment:
Probably originally my mistake, but this could use a brief javadoc.

Reply:
Updated.